tetris.py

```diff
@@ -125,38 +124,47 @@ def new_block(self):
             self.current_block_shape = self.upcoming_block_shape
         self.current_block_center = (int(game_width / 2), -2)
         self.upcoming_block_shape = random.choice(shape_names)
+        self.index = 0
 
-    def get_current_block(self):
+    def get_current_block(self, index):
         (x, y) = self.current_block_center
         if self.current_block_shape == "I":
-            return [(x - 2, y), (x - 1, y), (x, y), (x + 1, y)]
+            self.I = [[(x - 2, y), (x - 1, y), (x, y), (x + 1, y)], [(x, y-3), (x, y-2), (x, y-1), (x, y)]]
+            return self.I[self.index]
```

Comment:
Two things:
1. Because this variable is not used anywhere else, it should be a local variable, not something in `self`.
2. You are not using `%` like we discussed.

```suggestion
            I = [[(x - 2, y), (x - 1, y), (x, y), (x + 1, y)], [(x, y-3), (x, y-2), (x, y-1), (x, y)]]
            return I[self.index % len(I)]
```

Other parts of this function need a similar change.